doc/sphinx/proof-engine/ltac.rst

```diff
@@ -268,100 +244,121 @@ Goal selectors
-      for ``n-n`` when specifying multiple ranges.
+Selectors can also be used nested within a tactic expression with the
+:tacn:`only` tactic:
 
-   .. tacv:: all: @ltac_expr
-      :name: all: ...
+.. tacn:: only @selector : @ltac_expr3
 
-      In this variant, :token:`ltac_expr` is applied to all focused goals. ``all:`` can only
-      be used at the toplevel of a tactic expression.
+   .. insertprodn selector range_selector
 
-   .. tacv:: !: @ltac_expr
+   .. prodn::
+      selector ::= {+, @range_selector }
+      | [ @ident ]
+      range_selector ::= @num - @num
+      | @num
 
-      In this variant, if exactly one goal is focused, :token:`ltac_expr` is
-      applied to it. Otherwise the tactic fails. ``!:`` can only be
-      used at the toplevel of a tactic expression.
+   Applies :token:`ltac_expr3` to the selected goals.
 
-   .. tacv:: par: @ltac_expr
-      :name: par: ...
+   :tacn:`only` is an :token:`ltac_expr3`.
 
-      In this variant, :n:`@ltac_expr` is applied to all focused goals in parallel.
-      The number of workers can be controlled via the command line option
-      ``-async-proofs-tac-j`` taking as argument the desired number of workers.
-      Limitations: ``par:`` only works on goals containing no existential
-      variables and :n:`@ltac_expr` must either solve the goal completely or do
-      nothing (i.e. it cannot make some progress). ``par:`` can only be used at
-      the toplevel of a tactic expression.
+   :n:`{+, @range_selector }`
+      The selected goals are the union of the specified :token:`range_selector`\s.
 
-   .. exn:: No such goal.
-      :name: No such goal. (Goal selector)
-      :undocumented:
+   :n:`[ @ident ]`
+      Limits the application of :token:`ltac_expr3` to the goal previously named :token:`ident`
+      by the user (see :ref:`existential-variables`).
+
+   :n:`@num__1 - @num__2`
+      Selects the goals :n:`@num__1` through :n:`@num__2`, inclusive.
 
-   .. TODO change error message index entry
+   :n:`@num`
+      Selects a single goal.
+
+.. exn:: No such goal.
+   :name: No such goal. (Goal selector)
+   :undocumented:
 
-For loop
-~~~~~~~~
+.. TODO change error message index entry
 
-There is a for loop that repeats a tactic :token:`num` times:
+Do loop
+~~~~~~~
 
-.. tacn:: do @num @ltac_expr
+.. tacn:: do @int_or_var @ltac_expr3
    :name: do
 
+   The do loop repeats a tactic :token:`num` times:
+
    :n:`@ltac_expr` is evaluated to ``v`` which must be a tactic value. This tactic
    value ``v`` is applied :token:`num` times. Supposing :token:`num` > 1, after the
    first application of ``v``, ``v`` is applied, at least once, to the generated
-   subgoals and so on. It fails if the application of ``v`` fails before the num
+   subgoals and so on. It fails if the application of ``v`` fails before :token:`num`
    applications have been completed.
 
+   :tacn:`do` is an :token:`ltac_expr3`.
```

Comment:
If you do decide to apply the previous suggestion of referring to `l3_tactics`, then this and the rest of such sentences should be modified as well.